src/base/RadioCheckboxBase.js

```diff
@@ -39,30 +40,26 @@ function imgValidator (props, propName) {
   const asset = props[propName]
 
   if (!asset) {
-    return new Error(`The asset "${propName}" is required.`)
-  }
-
-  if (asset.url && !asset.url.includes('.svg')) {
-    return new Error('Image type is not supported.')
+    return new Error(`The background image "${propName}" is required.`)
   }
 }
 
-class RadioCheckboxBase extends Component {
+class RadioCheckboxBase extends React.PureComponent {
   static propTypes = {
     aria          : PropTypes.shape({
       label         :PropTypes.string,
     }),
-    assets        : PropTypes.shape({
+    bkgSvg        : PropTypes.shape({
       base          : imgValidator,
-      checked       : imgValidator,
+      selected      : imgValidator,
       disabled      : imgValidator,
     }),
     btnType       : PropTypes.oneOf(['radio', 'checkbox']).isRequired,
     isDisabled    : PropTypes.bool,
     children      : PropTypes.string,
     id            : PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
     isSelected    : PropTypes.bool,
-    onClick       : PropTypes.func,
+    onChange      : PropTypes.func,
```

Comment:
The reason for this prop change is that React kept complaining that when setting the checked attribute on the input tag, an onChange callback also had to be defined.